aptos-move/block-executor/src/executor.rs

```diff
@@ -348,8 +352,14 @@ where
                     Resource => versioned_cache.data().mark_estimate(&k, txn_idx),
                     Module => versioned_cache.modules().mark_estimate(&k, txn_idx),
                     Group => {
-                        versioned_cache.data().mark_estimate(&k, txn_idx);
+                        // Validation for both group size and metadata is based on values.
+                        // Execution may wait for estimates.
                         versioned_cache.group_data().mark_estimate(&k, txn_idx);
+
+                        // We are not marking metadata change as estimate, but after
```

Comment:
I was confused at first by the comment, maybe mention that the metadata lives in data versioned cache and so we do not have to nark it? Otherwise seems only clear when I see the diff with `versioned_cache.data().mark_estimate(&k, txn_idx);` removed 😄